lib/engine/tile.rb

```diff
@@ -360,6 +360,43 @@ def compute_city_town_edges
       ct_edges
     end
 
+    # this is invariant over rotations
+    def crossover?
+      return @_crossover if defined?(@_crossover)
+
+      @_crossover = compute_crossover
+    end
+
+    def compute_crossover
+      edge_paths = Hash.new { |h, k| h[k] = [] }
+      @paths.each do |p|
+        edge_paths[p.a.num].append(p) if p.a.edge?
+        edge_paths[p.b.num].append(p) if p.b.edge?
+        next unless p.nodes.size == 1
+
+        ct_edge = preferred_city_town_edges[p.nodes.first]
+        p.straight?(ct_edge)
+        p.gentle_curve?(ct_edge)
+      end
+
+      @paths.each do |p|
+        next if p.nodes.size > 2
```

Comment:
how can a path have more than two nodes?

Reply:
Damn. Should be 1.